libs/net/dal/Services/FileReferenceService.cs

```diff
@@ -23,9 +24,11 @@ public FileReferenceService(
         ClaimsPrincipal principal,
         IServiceProvider serviceProvider,
         StorageOptions options,
+        S3Options s3Options,
```

Comment:
Here you are using the singleton.  Is there a reason for this?

Reply:
This is for configuration only.  Use the standard `IOptions<S3Options>` approach.  If you need a singleton for connecting to S3, then you create a helper class and configure it through dependency injection as a singleton.